packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js

```diff
@@ -0,0 +1,101 @@
+import asyncStates from '@zooniverse/async-states'
+import React from 'react'
+import PropTypes from 'prop-types'
+import request from 'superagent'
+
+import LightCurveViewer from './LightCurveViewer'
+import locationValidator from '../../helpers/locationValidator'
+
+class LightCurveViewerContainer extends React.Component {
+  constructor () {
+    super()
+    
+    this.state = {
+      loading: asyncStates.initialized,
+      jsonData: null,
+    }
+    
+    //TODO: turn into variables?
+    this.width = 500
+    this.height = 500
+  }
+  
+  componentDidMount () {
+    if (this.props.subject) {
+      this.handleSubject()
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    const prevSubject = prevProps.subject
+    const { subject } = this.props
+
+    if (subject && (!prevSubject || prevSubject.id !== subject.id)) {
+      this.handleSubject()
+    }
+  }
+  
+  componentWillUnmount () {}
+
+  async handleSubject () {
+    const { subject } = this.props
+    
+    //Sanity check
+    //TODO: error handling - what if there's no subject?
+    if (!subject) return
+    
+    //Find the first location that has a JSON MIME type.
+    //NOTE: we also temporarily accept plain text, due to quirks with the Panoptes CLI uploading wonky MIME types (@shaun 20181024)
+    const jsonLocation = subject.locations.find(l => l['application/json'] || l['text/plain'])
```

Comment:
You could return the location value as a string, so you can pass it straight into the `request()` below.

Reply:
I've changed this to
```
let jsonLocation = subject.locations.find(l => l['application/json'] || l['text/plain'])
jsonLocation = jsonLocation && (jsonLocation['application/json'] || jsonLocation['text/plain'])
```
but I can't get it any more elegant that, since I need to factor for how `subject.locations.find()` might return `undefined` instead of a valid object.

For what it's worth,
```
let jsonLocation = subject.locations.filter(l =>l['application/json'] || l['text/plain']).map(l=>l['application/json'] || l['text/plain'])[0]
```
works too but that's hella harder to grok.